reflect.go

```diff
@@ -223,6 +223,16 @@ func (r *Reflector) reflectTypeToSchema(definitions Definitions, t reflect.Type)
 		return &Type{Ref: "#/definitions/" + r.typeName(t)}
 	}
 
+	if r.TypeMapper != nil {
+		if t := r.TypeMapper(t); t != nil {
+			return t
+		}
+	}
+
+	if rt := r.reflectCustomType(definitions, t); rt != nil {
```

Comment:
There is small change to existing generation: because `reflectCustomType` is now called on structs which are in `IgnoredTypes`, structs with `JSONSchemaType` cannot be listed in `IgnoredTypes`.